### ISSUE TITLE
read_proj_geno produces old gcl format

### ISSUE THREAD

**Issue by csjalbert** [commfish/GCLr#17]
@krshedd and @awbarclay
I'm trying to modernize the QC script and realized `read_proj_geno()` produces the old style (i.e., list) gcl format. I assume this is just for compatibility with the QC script but is there any other reason why we shouldn't update `read_proj_geno()` to the new style? We could use `old2new` to convert the object for QC purposes, but I'd rather fix the function.

**Comment by csjalbert** [comment]
I don't think we can't use `old2new` because of the "slim" gcl object. I'm still looking at the issue with `old2new`, but, this is one of the functions that generates it's own locus control. @awbarclay, was this on your list of functions to edit? If so, let's reassess the whole thing before you start working on it.

**Comment by krshedd** [comment]
@csjalbert 
Correct, `read_proj_geno()` was maintained in the old style 4-part list format for compatibility with all the other QC-specific functions such as `failure_rate`.  I think we should just update `read_proj_geno()` rather than try to use `old2new`.

Speaking of modernizing QC, it looks like `dupcheck_among_sillys` already got converted over to new style tibble format? If so, we are pretty close!

**Comment by awbarclay** [comment]
Yes, `read_proj_geno()` is one that I worked on already.  I agree with Kyle, I think we should just update the function rather than use `old2new` to convert the objects.  I already updated `dupcheck_among_sillys` because it's used outside of QC analyses.

**Comment by csjalbert** [comment]
This turned into a larger issue. I manually generated the slim gcl object to get around the `read_proj_geno()` issue but `failure_rate()` failed. There are a few issues with this function:
1) it needs to be updated to work with the new gcl format, 
2) `pool_collections()` is used within but that does not work with a slim gcl object

The `pool_collections()` issue has major downstream impacts, so I wanted to run it by everyone before making any changes.

@krshedd had a few quick thoughts on the `pool_collections` problem but we haven't thought though any of them:

1. if `pool_collections` doesn't show up much in QC, just add the code for pooling a slim object within `failure_rate`
2. write a `pool_collections_qc` or `pool_collections_slim`
3. update `pool_collections` with an if/else to look for slim objects

I was originally thinking something like number 3, but am open to other ideas. 

In short, QC is broken. We need to fix `read_proj_geno`,  `failure_rate`, `pool_collections`, and likely others (e.g., `dupcheck_within_silly`, which also has `attr` assigned then selected )

**Comment by csjalbert** [comment]
I'm not sure I'll have time to finalize testing until next week but my new idea is to simply add the extra attributes as `NAs`. We can just add a section to the end of of `read_proj_geno` that adds the missing attributes. That should allow for any downstream functions to work without modification. I also propose a message along the lines of _this is a slim gcl object with additional attributes added as NAs for compatibility with downstream functions._

**Comment by csjalbert** [comment]
Adding the additional attributes as `NA` worked. Closing and documenting as fixed in commit f69b3df.
